config/config.go

```diff
@@ -560,6 +577,14 @@ func defaultUnmarshaler(componentViperSection *viper.Viper, intoCfg interface{})
 	return componentViperSection.UnmarshalExact(intoCfg)
 }
 
+func validator(config configmodels.Receiver) (configmodels.CustomValidator, error) {
+	cfg, ok := config.(configmodels.ConfigValidator)
+	if !ok {
+		return nil, errors.New("validate interface is not implemented by Config")
+	}
+	return cfg.Validate, nil
+}
```

Comment:
Why don't we make the `ConfigValidator` interface optional and do here something similar to what we do for unmarshaling?
https://github.com/open-telemetry/opentelemetry-collector/blob/1b164905a92c4c94ece83f66c58250f0531a2be1/config/config.go#L569-L574

We can have a `defaultUnmarshaler` that just does `return nil`.

Reply:
We discussed this, and we believe that for validate we want users to be forced to implement it, so they are forced to think about that. Also most of the components implementation that I've seen require some validation which sometimes happens in constructor or sometimes in start (very inconsistent).

For the "unmarshaler" that is a very edge case so going with a separate interface makes sense in that case. Does it make sense?

Reply:
Makes sense, yes, thanks! I wasn't aware of that discussion but I agree with your points

Reply:
I think was an ad-hoc meeting where we discussed that, sorry for not sharing it. Maybe @mxiamxia you can add this to the PR description and explain why Validate is a mandatory thing to have.

Reply:
Explained why should we have a mandatory `Validate()` in the PR description. Thx!